unleash-yggdrasil/src/strategy_parsing.rs

```diff
@@ -353,14 +357,30 @@ fn list_constraint(inverted: bool, mut node: Pairs<Rule>) -> RuleFragment {
 
                 match comparator {
                     ContentComparator::In => match context_value {
-                        Some(context_value) => values.contains(&context_value).invert(inverted),
+                        Some(context_value) => values.contains(&context_value),
                         None => false,
                     },
                     ContentComparator::NotIn => match context_value {
-                        Some(context_value) => !values.contains(&context_value).invert(inverted),
+                        Some(context_value) => !values.contains(&context_value),
+                        None => true,
+                    },
+                    ContentComparator::InIgnoreCase => match context_value {
+                        Some(context_value) => {
+                            let needle = context_value.to_lowercase();
+                            values.iter().find(|x| x.to_lowercase() == needle).is_some()
+                        }
+                        None => false,
+                    },
+
+                    ContentComparator::NotInIgnoreCase => match context_value {
+                        Some(context_value) => {
+                            let needle = context_value.to_lowercase();
+                            values.iter().find(|x| x.to_lowercase() == needle).is_none()
+                        }
                         None => true,
                     },
                 }
+                .invert(inverted)
```

Comment:
Not sure if this is standard in rust, or if it's better to keep everything together. Lines started to grow big and this little change made it more readable, but I'm ok moving it back (also for consistency with the other operations)

Reply:
Unfortunately that's a logic change here, since this will also flip the state when the context value doesn't exist. 

Annoying that neither the existing tests nor the client spec tests picked it up. 

This might be more correct but I think it needs a bit more thought than a 5 minute glance